File: src/main/java/org/railwaystations/rsapi/adapter/in/web/controller/StatisticController.java
package org.railwaystations.rsapi.adapter.in.web.controller;

import lombok.RequiredArgsConstructor;
import org.railwaystations.rsapi.adapter.in.web.api.StatisticApi;
import org.railwaystations.rsapi.adapter.in.web.model.StatisticDto;
import org.railwaystations.rsapi.core.model.Statistic;
import org.railwaystations.rsapi.core.ports.in.GetStatisticUseCase;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class StatisticController implements StatisticApi {

    private final GetStatisticUseCase getStatisticUseCase;

    private StatisticDto getStatistic(String country) {
        return toDto(getStatisticUseCase.getStatistic(country));
    }

    private StatisticDto toDto(Statistic statistic) {
        return new StatisticDto(statistic.total(), statistic.withPhoto(), statistic.withoutPhoto(), statistic.photographers())
                .countryCode(statistic.countryCode());
    }

    @Override
    public ResponseEntity<StatisticDto> statsGet(String country) {
        return ResponseEntity.ok(getStatistic(country));
    }
}

File: adapter/src/main/kotlin/org/railwaystations/rsapi/adapter/web/controller/StatisticController.kt
package org.railwaystations.rsapi.adapter.web.controller

import org.railwaystations.rsapi.adapter.web.api.StatisticApi
import org.railwaystations.rsapi.adapter.web.model.StatisticDto
import org.railwaystations.rsapi.core.model.Statistic
import org.railwaystations.rsapi.core.ports.inbound.GetStatisticUseCase
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RestController

@RestController
class StatisticController(private val getStatisticUseCase: GetStatisticUseCase) : StatisticApi {

    override fun getStats(country: String?): ResponseEntity<StatisticDto> {
        return ResponseEntity.ok(getStatisticUseCase.getStatistic(country).toDto())
    }

}

private fun Statistic.toDto() = StatisticDto(
    total = total.toLong(),
    withPhoto = withPhoto.toLong(),
    withoutPhoto = withoutPhoto.toLong(),
    photographers = photographers.toLong(),
    countryCode = countryCode
)
